Lib/pprint.py

```diff
@@ -525,6 +527,13 @@ def _safe_repr(self, object, context, maxlevels, level):
             return repr(object), True, False
 
         r = getattr(typ, "__repr__", None)
+
+        if issubclass(typ, int):
```

Comment:
```suggestion
        if issubclass(typ, int) and r is int.__repr__:
```

An example of where this matters:

```python
class DollarAmount(int):
    def __new__(cls, string):
        dollars, _, cents = string.lstrip("$").partition(".")
        total_cents = int(dollars) * 100 + int(cents)
        return super().__new__(DollarAmount, total_cents)
    def __repr__(self):
        q, r = divmod(self, 100)
        return f"{self.__class__.__qualname__}('${q}.{r}')"


print(DollarAmount('$19.95'))
pprint.pprint(DollarAmount('$19.95'))
# should both print DollarAmount('$19.95'), not 1_995
```

That might deserve a test case.

Reply:
Fixed. I added a simpler test case to improve code coverage too.

Thanks for the review!